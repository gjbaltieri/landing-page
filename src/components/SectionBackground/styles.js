import styled, { css } from 'styled-components';

const containerBackground = (theme) =>
  css`
    background: ${theme.color.background.black};
    color: ${theme.color.heading.white};
  `;

export const Container = styled.div`
  ${({ theme, background }) => css`
    transition: background 300ms ease-in-out;
    background: ${theme.color.background.white};
    color: ${theme.color.black};
    min-height: 100vh;
    width: 100%;
    ${background && containerBackground(theme)};
  `}
`;
