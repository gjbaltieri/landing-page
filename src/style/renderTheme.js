import { render } from '@testing-library/react';
import { GlobalStyles } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}
