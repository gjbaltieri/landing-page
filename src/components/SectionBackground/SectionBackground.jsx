import P from 'prop-types';
import * as Styled from './styles';

export default function SectionBackground({ children, background = false }) {
  return (
    <Styled.Container background={background}>{children}</Styled.Container>
  );
}

SectionBackground.propTypes = {
  children: P.node.isRequired,
  background: P.bool,
};
