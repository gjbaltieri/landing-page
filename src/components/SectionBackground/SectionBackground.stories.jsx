import React from 'react';
import SectionBackground from './SectionBackground';
import SectionContainer from '../SectionContainer/SectionContainer';

export default {
  title: 'SectionBackground',
  component: SectionBackground,
  args: {
    children: (
      <SectionContainer>
        <div>
          <h1>Section Container</h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quidem
            consectetur odit neque, qui ad at enim cum velit omnis aliquam
            corrupti quisquam ab quis voluptatum totam rerum ea eaque iste!
          </p>
        </div>
      </SectionContainer>
    ),
  },
  argTypes: {
    children: {
      control: {
        type: '',
      },
    },

    background: { type: 'boolean' },
  },
};

export const Template = (args) => (
  <div>
    <SectionBackground {...args} />
  </div>
);
