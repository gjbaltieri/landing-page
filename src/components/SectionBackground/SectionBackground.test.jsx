import 'jest-styled-components';
import '@testing-library/jest-dom';
import renderTheme from '../../style/renderTheme';
import SectionBackground from './SectionBackground';

describe('<SectionBackground/>', () => {
  test('should render SectionBackground with background dark', () => {
    const { container } = renderTheme(
      <SectionBackground background={true}>
        <h1>Children</h1>
      </SectionBackground>
    );
    expect(container).toMatchSnapshot();
  });

  test('should render SectionBackground with background white', () => {
    const { container } = renderTheme(
      <SectionBackground background={false}>
        <h1>Children</h1>
      </SectionBackground>
    );
    expect(container).toMatchSnapshot();
  });
});
