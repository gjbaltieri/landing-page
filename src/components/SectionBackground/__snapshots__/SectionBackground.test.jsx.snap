// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground/> should render SectionBackground with background dark 1`] = `
.c0 {
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
  background: #F2F2F2;
  min-height: 100vh;
  width: 100%;
  background: #0D0D0D;
  color: #F2F2F2;
}

.c1 {
  max-width: 120rem;
  padding: 2.4rem;
  width: 100%;
  margin: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<SectionBackground/> should render SectionBackground with background white 1`] = `
.c0 {
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
  background: #F2F2F2;
  min-height: 100vh;
  width: 100%;
}

.c1 {
  max-width: 120rem;
  padding: 2.4rem;
  width: 100%;
  margin: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;
